packages/extension-api/src/extension-api.d.ts

```diff
@@ -1330,6 +1330,18 @@ declare module '@podman-desktop/api' {
      * @param manifests the manifests to create as JSON objects
      */
     export function createResources(context: string, manifests: unknown[]): Promise<void>;
+
+    /**
+     * Add a KubernetesGenerator to KubernetesGeneratorRegistry
+     * @param kubeGenerator the generator to add
+     */
+    export function registerKubernetesGeneratorProvider(kubeGenerator: KubeGenerator): Disposable;
+
+    /**
+     * Remove a provider from the KubernetesGeneratorRegistry
+     * @param kubeGenerator the generator to remove
+     */
+    export function unregisterKubernetesGeneratorProvider(kubeGenerator: KubeGenerator): void;
```

Comment:
you can delete the unregister

as register returns a disposable resource, when calling `dispose()` it'll unregister the provider